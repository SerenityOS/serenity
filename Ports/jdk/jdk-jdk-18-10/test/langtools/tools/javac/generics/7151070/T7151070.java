/*
 * @test /nodynamiccopyright/
 * @bug     7151070
 * @summary NullPointerException in Resolve.isAccessible
 * @compile/fail/ref=T7151070.out -XDrawDiagnostics T7151070.java
 */

class T7151070a {
    private static class PrivateCls { }
    public static class PublicCls extends PrivateCls { }

    public void m(PrivateCls p) { }
}

class T7151070b {
    public void test(Test<T7151070a.PublicCls> obj, T7151070a outer) {
        outer.m(obj.get());
    }

    public static class Test<T> {
        public T get() {
            return null;
        }
    }
}
