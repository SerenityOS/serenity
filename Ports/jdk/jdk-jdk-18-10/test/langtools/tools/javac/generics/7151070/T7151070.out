T7151070.java:17:24: compiler.err.report.access: T7151070a.PrivateCls, private, T7151070a
1 error
